resources/snakefiles/Snakefile.smk

```diff
@@ -0,0 +1,178 @@
+    params: ""
+    threads:
+        config['threads']['fastqc_post_trim']
+    wrapper:
+        "0.72.0/bio/fastqc"
+
+
+rule host_filter:
+    """
+    Performs host read filtering on paired end data using Bowtie and Samtools/
+    BEDtools.
+
+    Also requires an indexed reference (path specified in config).
+
+    First, uses Bowtie output piped through Samtools to only retain read pairs
+    that are never mapped (either concordantly or just singly) to the indexed
+    reference genome. Fastqs from this are gzipped into matched forward and
+    reverse pairs.
+
+    Unpaired forward and reverse reads are simply run through Bowtie and
+    non-mapping gzipped reads output.
+
+    All piped output first written to localscratch to avoid tying up filesystem.
+    """
+    input:
+        fastq1=rules.cutadapt_pe.output.fastq1,
+        fastq2=rules.cutadapt_pe.output.fastq2
+    output:
+        nonhost_R1="output/filtered/nonhost/{sample}.1.fastq.gz",
+        nonhost_R2="output/filtered/nonhost/{sample}.2.fastq.gz",
+        host="output/filtered/host/{sample}.bam",
+        temp_dir=temp(directory("output/{sample}_temp"))
+    params:
+        ref=config['host_reference']
+    conda:
+        "resources/envs/bowtie2.yaml"
+    threads:
+        config['threads']['host_filter']
+    benchmark:
+        "output/benchmarks/bowtie2/{sample}_benchmark.txt"
+    log:
+        "output/logs/bowtie2/{sample}.bowtie.log"
+    shell:
+        """
+        # Make temporary and permanent output directories
+        mkdir -p "output/filtered/host/"
+        mkdir -p "output/filtered/nonhost/"
+        mkdir -p {output.temp_dir}
+
+        # Map reads against reference genome
+        bowtie2 -p {threads} -x {params.ref} \
+          -1 {input.fastq1} -2 {input.fastq2} \
+          --un-conc-gz {wildcards.sample}_nonhost \
+          2> {log} | samtools view -bS - > {output.host}
+
+        # rename nonhost samples
+        mv {wildcards.sample}_nonhost.1 output/filtered/nonhost/{wildcards.sample}.1.fastq.gz
+        mv {wildcards.sample}_nonhost.2 output/filtered/nonhost/{wildcards.sample}.2.fastq.gz
+        """
+
+rule metaspades_assembly:
+    """
+
+    Performs a metagenomic assembly on a sample using MetaSPAdes.
+
+    """
+    input:
+        fastq1=rules.host_filter.output.nonhost_R1,
+        fastq2=rules.host_filter.output.nonhost_R2
+    output:
+        contigs="output/metaspades/{sample}.contigs.fasta",
+        temp_dir=temp(directory("output/{sample}_temp"))
+    conda:
+        "resources/envs/spades.yaml"
+    threads:
+        config['threads']['assembly']
+    benchmark:
+        "output/benchmarks/metaspades/{sample}_benchmark.txt"
+    log:
+        "output/logs/metaspades/{sample}.metaspades.log",
+    shell:
+        """
+        # Make temporary output directory
+        mkdir -p {output.temp_dir}
+
+        # Make directory for assemblies and log files
```

Comment:
You shouldn't need to make these directories manually -- SNakemake will create them if they don't exist at the start of rule execution.

Reply:
done!